### PR TITLE
Improve aggregation selection verification

### PR DESCRIPTION
**Pull request by ericanderson**
Mismatch IDE:
image

Hover over `mismatchedLatitude`:
image

Hover over `testAggregateCountWithGroups`:
image

### CODE REVIEW

File: packages/client/src/objectSet/createObjectSet.ts [hunk 1]
```diff
@@ -63,20 +63,30 @@ export function createObjectSet<Q extends ObjectOrInterfaceDefinition>(
...
+      },
     ): Promise<AggregationsResults<Q, AO>> => {
       return aggregateOrThrow<Q, AO>(
         clientCtx,
         objectType,
         objectSet,
         req,
       );
-    },
+    }) as any,
```
Comment (ssanjay1) [inline packages/client/src/objectSet/createObjectSet.ts:89]: Why do we need the as any here?

File: packages/client/src/object/aggregateOrThrow.ts [hunk 2]
```diff
@@ -111,3 +121,5 @@ function aggregationToCountResult(
     }
   }
 }
+
```
Comment (ssanjay1) [inline packages/client/src/object/aggregateOrThrow.ts:124]: Was this intentional?

**Review by ssanjay1 — APPROVED** [review]

File: packages/client/src/objectSet/createObjectSet.ts (hunk 1 above)
Comment (ericanderson) [reply packages/client/src/objectSet/createObjectSet.ts:89]: I can try to take it out but the types got so complex here that it was erroring that two identical types didnt match.

Comment (ssanjay1) [reply packages/client/src/objectSet/createObjectSet.ts:89]: Ah probably fine to just keep it then, things work so yay

File: packages/client/src/query/aggregations/AggregationsResults.ts [hunk 3]
```diff
@@ -25,12 +26,17 @@ import type {
...
   : Exclude<AO["groupBy"], undefined> extends never // groupBy is explicitly undefined
     ?
       & AggregationResultsWithoutGroups<Q, AO["select"]>
       & AggregationCountResult<Q, AO["select"]>
-  : AggregationResultsWithGroups<Q, AO["select"], AO["groupBy"]>;
+  : AggregationResultsWithGroups<Q, AO["select"], AO["groupBy"]>
+  : `Sorry, the following are not valid selectors for an aggregation: ${Exclude<
+    keyof AO["select"] & string,
+    AggregatableKeys<Q> | "$count"
+  >}`;
```
Comment (mfedderly) [inline packages/client/src/query/aggregations/AggregationsResults.ts:42]: I kinda dig this way of surfacing more useful errors.

**Review by mfedderly — APPROVED** [review]

**Pull request merged**